src/services/transpile.ts

```diff
@@ -135,14 +195,18 @@ export function transpileModule(input: string, transpileOptions: TranspileOption
     let outputText: string | undefined;
     let sourceMapText: string | undefined;
 
-    const program = createProgram([inputFileName], options, compilerHost);
+    const inputs = declaration ? [inputFileName, barebonesLibName] : [inputFileName];
+    const program = createProgram(inputs, options, compilerHost);
 
     if (transpileOptions.reportDiagnostics) {
         addRange(/*to*/ diagnostics, /*from*/ program.getSyntacticDiagnostics(sourceFile));
         addRange(/*to*/ diagnostics, /*from*/ program.getOptionsDiagnostics());
     }
     // Emit
-    program.emit(/*targetSourceFile*/ undefined, /*writeFile*/ undefined, /*cancellationToken*/ undefined, /*emitOnlyDtsFiles*/ undefined, transpileOptions.transformers);
+    const result = program.emit(/*targetSourceFile*/ undefined, /*writeFile*/ undefined, /*cancellationToken*/ undefined, /*emitOnlyDtsFiles*/ declaration, transpileOptions.transformers, /*forceDtsEmit*/ declaration);
+
+    // TODO: Should this require `reportDiagnostics`?

```

Comment:
Is it the case that the JS emit pipeline never reports errors?

Reply:
Correct - emit pipeline errors were added for normal declaration emit when we ported them to transforms (and they add visibility errors _very late_) and are unused by js emit transforms -`isolatedDeclarations` continues to use the mechanism extensively. It's a longstanding issue I've wanted to fix, since it means we can't get declaration emit errors without _actually running declaration emit_ (remind yourself how `getPreEmitDiagnostics` works some time 😑), but it's _super annoying_ to try and decouple the errors from the node generation step.